### PR TITLE
feat: get social links for erc20 token

### PR DESCRIPTION
**Pull request by Lanttcat**

### CODE REVIEW

File: src/providers/coingecko.ts [hunk 1]
```diff
@@ -34,7 +36,101 @@ interface Coin {
...
+  id: string
+  symbol: string
+  name: string
+  asset_platform_id: any
+  platforms: {
+    '': string
+  }
+  detail_platforms: {
+    '': {
+      decimal_place: any
```
Comment (guanbinrui) [inline src/providers/coingecko.ts]: Avoid any.

File: src/providers/coingecko.ts [hunk 2]
```diff
@@ -34,7 +36,101 @@ interface Coin {
...
+      contract_address: string
+    }
+  }
+  block_time_in_minutes: number
+  hashing_algorithm: string
+  categories: Array<string>
```
Comment (guanbinrui) [inline src/providers/coingecko.ts]: string[]

Comment (guanbinrui) [inline src/providers/coingecko.ts]: string[]

File: src/providers/coingecko.ts [hunk 3]
```diff
@@ -34,7 +36,101 @@ interface Coin {
...
+    chat_url: Array<string>
+    announcement_url: Array<string>
+    twitter_screen_name: string
+    facebook_username: string
+    bitcointalk_thread_identifier: any
+    telegram_channel_identifier: string
+    subreddit_url: string
+    repos_url: {
+      github: Array<string>
+      bitbucket: Array<any>
```
Comment (guanbinrui) [inline src/providers/coingecko.ts]: Avoid any.

File: src/providers/coingecko.ts [hunk 4]
```diff
@@ -34,7 +36,101 @@ interface Coin {
...
+      telegram: coin.links.telegram_channel_identifier,
+    }
+  }
+
+  private async getMetadata(ids: (string | number)[]) {
+    const result: any = {}
+    for (const id of ids) {
+      try {
+        const metadataURL = urlcat(baseProURL, '/api/v3/coins/:id', {
+          id: id,
```
Comment (guanbinrui) [inline src/providers/coingecko.ts]: id,

**Review by hanszhoou — APPROVED** [review]

File: src/providers/coinmarketcap.ts [hunk 5]
```diff
@@ -29,17 +29,48 @@ export interface Platform {
   token_address: string
 }
 
+export type CoinDetail = {
+  urls: {
+    website: Array<string>
+    technical_doc: Array<string>
+    twitter: Array<any>
```
Comment (guanbinrui) [inline src/providers/coinmarketcap.ts]: Avoid `any`.

File: src/providers/coinmarketcap.ts [hunk 6]
```diff
@@ -29,17 +29,48 @@ export interface Platform {
   token_address: string
 }
 
+export type CoinDetail = {
+  urls: {
+    website: Array<string>
```
Comment (guanbinrui) [inline src/providers/coinmarketcap.ts]:
```suggestion
    website: string[]
```

File: src/providers/coinmarketcap.ts [hunk 7]
```diff
@@ -29,17 +29,48 @@ export interface Platform {
...
+  logo: string
+  id: number
+  name: string
+  symbol: string
+  slug: string
+  description: string
+  date_added: string
+  date_launched: string
+  tags: Array<string>
+  platform: any
```
Comment (guanbinrui) [inline src/providers/coinmarketcap.ts]: Avoid `any`.

File: src/type.ts [hunk 8]
```diff
@@ -94,6 +94,17 @@ export enum SourceType {
   R2D2 = 'R2D2',
 }
 
+export type SocialLinks=  {
```
Comment (guanbinrui) [inline src/type.ts]: ?

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**